### PR TITLE
Allow UDF reloading for streaming templates

### PR DESCRIPTION
**Pull request by fozzie15**

### CODE REVIEW

File: v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToBigQuery.java [hunk 1]
```diff
@@ -79,6 +79,8 @@ public static void main(String[] args) {
             TransformTextViaJavascript.newBuilder()
                 .setFileSystemPath(options.getJavascriptTextTransformGcsPath())
                 .setFunctionName(options.getJavascriptTextTransformFunctionName())
+                .setReloadFunction(options.getJavascriptFunctionReload())
+                .setReloadIntervalMinutes(options.getJavascriptReloadIntervalMinutes())
```
Comment (bvolpato) [inline v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToBigQuery.java:83]: I don't think it makes a lot of sense for batch pipelines

File: v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToDatastoreDelete.java [hunk 2]
```diff
@@ -126,6 +126,8 @@ public static void main(String[] args) {
             TransformTextViaJavascript.newBuilder()
                 .setFileSystemPath(options.getJavascriptTextTransformGcsPath())
                 .setFunctionName(options.getJavascriptTextTransformFunctionName())
+                .setReloadFunction(options.getJavascriptFunctionReload())
+                .setReloadIntervalMinutes(options.getJavascriptReloadIntervalMinutes())
```
Comment (bvolpato) [inline v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToDatastoreDelete.java:130]: batch

File: v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToPubsub.java [hunk 3]
```diff
@@ -59,6 +59,8 @@ public static void main(String[] args) {
             TransformTextViaJavascript.newBuilder()
                 .setFileSystemPath(options.getJavascriptTextTransformGcsPath())
                 .setFunctionName(options.getJavascriptTextTransformFunctionName())
+                .setReloadFunction(options.getJavascriptFunctionReload())
+                .setReloadIntervalMinutes(options.getJavascriptReloadIntervalMinutes())
```
Comment (bvolpato) [inline v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToPubsub.java:63]: batch

File: v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToText.java [hunk 4]
```diff
@@ -106,6 +106,8 @@ public static void main(String[] args) {
             TransformTextViaJavascript.newBuilder()
                 .setFileSystemPath(options.getJavascriptTextTransformGcsPath())
                 .setFunctionName(options.getJavascriptTextTransformFunctionName())
+                .setReloadFunction(options.getJavascriptFunctionReload())
+                .setReloadIntervalMinutes(options.getJavascriptReloadIntervalMinutes())
```
Comment (bvolpato) [inline v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToText.java:110]: batch

File: v1/src/main/java/com/google/cloud/teleport/templates/TextIOToBigQuery.java [hunk 5]
```diff
@@ -144,6 +144,8 @@ public static void main(String[] args) {
             TransformTextViaJavascript.newBuilder()
                 .setFileSystemPath(options.getJavascriptTextTransformGcsPath())
                 .setFunctionName(options.getJavascriptTextTransformFunctionName())
+                .setReloadFunction(options.getJavascriptFunctionReload())
+                .setReloadIntervalMinutes(options.getJavascriptReloadIntervalMinutes())
```
Comment (bvolpato) [inline v1/src/main/java/com/google/cloud/teleport/templates/TextIOToBigQuery.java:148]: batch

File: v1/src/main/java/com/google/cloud/teleport/templates/TextToDatastore.java [hunk 6]
```diff
@@ -117,6 +117,8 @@ public static void main(String[] args) {
             TransformTextViaJavascript.newBuilder()
                 .setFileSystemPath(options.getJavascriptTextTransformGcsPath())
                 .setFunctionName(options.getJavascriptTextTransformFunctionName())
+                .setReloadFunction(options.getJavascriptFunctionReload())
+                .setReloadIntervalMinutes(options.getJavascriptReloadIntervalMinutes())
```
Comment (bvolpato) [inline v1/src/main/java/com/google/cloud/teleport/templates/TextToDatastore.java:121]: batch

File: v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToBigQuery.java (hunk 1 above)
Comment (bvolpato) [reply v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToBigQuery.java:83]: Explaining: batch jobs are short-lived by definition, I think having a single runtime for every run makes things simpler / predictable

File: v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToText.java (hunk 4 above)
Comment (fozzie15) [reply v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToText.java]: Done

File: v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToBigQuery.java (hunk 1 above)
Comment (fozzie15) [reply v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToBigQuery.java]: Removed

File: v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToDatastoreDelete.java (hunk 2 above)
Comment (fozzie15) [reply v1/src/main/java/com/google/cloud/teleport/templates/DatastoreToDatastoreDelete.java]: Done

File: v1/src/main/java/com/google/cloud/teleport/templates/TextIOToBigQuery.java (hunk 5 above)
Comment (fozzie15) [reply v1/src/main/java/com/google/cloud/teleport/templates/TextIOToBigQuery.java]: Done

File: v1/src/main/java/com/google/cloud/teleport/templates/TextToDatastore.java (hunk 6 above)
Comment (fozzie15) [reply v1/src/main/java/com/google/cloud/teleport/templates/TextToDatastore.java]: Done

**Review by bvolpato — APPROVED** [review]

**Pull request merged**